lib/smf/smf.c

```diff
@@ -9,24 +9,24 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(smf);
 
-/*
- * Private structure (to this file) used to track state machine context.
- * The structure is not used directly, but instead to cast the "internal"
- * member of the smf_ctx structure.
+/**
+ * @brief Private structure (to this file) used to track state machine context.
+ *        The structure is not used directly, but instead to cast the "internal"
+ *        member of the smf_ctx structure.
  */
 struct internal_ctx {
-	bool new_state : 1;
-	bool terminate : 1;
-	bool exit      : 1;
-	bool handled   : 1;
+	bool new_state: 1;
+	bool terminate: 1;
+	bool exit: 1;
+	bool handled: 1;
+	bool entry: 1;
```

Comment:
It appears that you only ever set the `entry` flag, but you never check it.
Was the intent to use this to prevent `smf_set_state` calls from an entry action?

Also, it would be clearer if these were named `is_exit` and `is_entry` to avoid confusion with the state's `exit` and `entry` function pointers.

Reply:
Yes it was. I'll delete it